titus-server-master/src/main/java/com/netflix/titus/master/jobmanager/service/KubeNotificationProcessor.java

```diff
@@ -449,4 +466,27 @@ private static Task addMissingAttributes(PodWrapper podWrapper, Task updatedTask
         updatedAllocation.ifPresent(id -> builder.addToTaskContext(TASK_ATTRIBUTES_OPPORTUNISTIC_CPU_ALLOCATION, id));
         return builder.build();
     }
+
+    /**
+     * 'updatedTask' is a modified copy of 'currentTask' (or exactly the same version of the object if nothing changed).
+     */
+    @VisibleForTesting
+    static boolean areTasksEquivalent(Task currentTask, Task updatedTask) {
+        if (currentTask == updatedTask) {
```

Comment:
Guessing this is just defensive programming? I don't think this case is possible though since the objects are created on demand every event.

Reply:
No, it is possible.